### ISSUE TITLE
Could CONTRIBUTING.md be improved?

### ISSUE THREAD

**Issue by mightyiam** [rust-lang/rust#77215]
The [CONTRIBUTING.md](https://github.com/rust-lang/rust/blob/85072e330394440776ace5ae612e3ced32ef6348/CONTRIBUTING.md), writes:

> # Contributing to Rust
>
> Thank you for your interest in contributing to Rust!
>
> To get started, read the [Getting Started] guide in the [rustc-dev-guide].

Perhaps it could be a little better. What throws me off in this last sentence, is the term `rustc`. I'm just a total newbie and just getting to know the terms in this language, so there's no surprise that I can't explain why I am sent one place or another for documentation. So, this is more of a question to those familiar with the constellation of terms, repositories, web sites, and so on. Could this reference explain why it refers to `rustc-dev-guide` and not anything else. I want to contribute to documentation of what seems to be a core macro (#77100). So, I'm not sure that reading the "Guide to Rustc Development" would be the correct thing to do (even thought I will surely personally read it at some point). The readme of this repository writes:

> This is the main source code repository for Rust. It contains the compiler, standard library, and documentation.

So, in the `CONTRIBUTING.md` file, I expect to be directed to instructions on contributing to all or any of the compiler, standard library and documentation. But I am directed to what seems to be — at least by title alone (I haven't read into it, yet) — solely about contributing to the rust compiler.

Any suggestion regarding how to improve this — I have the intention of implementing myself. By the way, I am streaming my contribution to rust live at https://twitch.tv/mightyiam.

**Comment by pickfire** [comment]
I personally think `CONTRIBUTING.md` should still redirect to rustc-dev-guide but I think the guide could be improved. IIRC the guide was not updated regarding the split of `compiler`, `library` and `core`. The guide talks more about contributing to the rust compiler, but personally I think contributing to the library should be straightforward besides not knowing how to build a custom toolchain which isn't always necessary. @mightyiam I believe you can just create a pull request for this if you see fit to close the issue (so it doesn't hang around and become lost).

**Comment by follower** [comment]
@mightyiam Yeah, I've very recently run into a similar issue; based on my research, it seems the original content of `CONTRIBUTING.md` now lives at https://rustc-dev-guide.rust-lang.org/contributing.html which seems like a better entry point.

It seems the contribution/developer docs are currently mid-transition from one approach to a new one.

Related:

 * https://github.com/rust-lang/rustc-dev-guide/issues/775#issuecomment-699063082

 * https://github.com/rust-lang/rust/pull/77195

**Comment by LeSeulArtichaut** [comment]
cc @rust-lang/wg-rustc-dev-guide

**Comment by follower** [comment]
Having just discovered that the GitHub "first issue" pop-up _also_ links to this document, I just went ahead & created PR #77268 which links to the chapter of the dev guide that now holds the content that was previously in `CONTRIBUTING.md`.

Until the "Getting Started" chapter gets re-written to be less `rustc`-specific, linking to it rather than "Contributing to Rust" chapter seems unnecessarily unwelcoming to new contributors, particularly those who were specifically directed there while creating an issue.

(Of course, if the "first issue" link can be customised that's another option for the future.)

**Comment by henryboisdequin** [comment]
@rusbot claim

**Comment by mightyiam** [comment]
Thank you.
